src/Container/HStringContains.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace Haystack\Container;
+
+use Haystack\Helpers\Helper;
+use Haystack\HString;
+
+class HStringContains
+{
+    /** @var string */
+    private $str;
+
+    /** @var string */
+    private $value;
+
+    public function __construct(HString $hString)
+    {
+        $this->str = $hString->toString();
+    }
+
+    /**
+     * @param HString|string $value
+     * @return bool
+     */
+    public function contains($value)
+    {
+        if ($value instanceof HString) {
+            $value = $value->toString();
+        }
+
+        if (is_scalar($value)) {
+            $this->value = (string) $value;
+        } else {
+            throw new \InvalidArgumentException(sprintf("%s is neither a scalar value nor an HString", Helper::getType($value)));
+        }
+
+        return $this->containsValue();
+    }
+
+    /**
+     * @return bool
+     */
+    private function containsValue()
+    {
+        $pos = strpos($this->str, $this->value);
+
+        return (false !== $pos) ?: false;
```

Comment:
[NB] I find the ternary operator use here makes this confusing. It's easy to reason about but I have to think a minute to be sure I've got it straight.

The true-condition block doesn't explicitly list a value `true`, yet has `false` in the false-condition block. The ternary isn't needed in this case.

That slight quick is compounded by the double-negative `false !==` part of the comparison. So here's my thought process when looking at this function...

"So when false is not equal to $pos... What's the $pos again?
Oh yeah the numeric position if the string is found... unless it's not found, then it's `false`, right?
OK then which part of the ternary operator are we returning...? The missing true-condition? Or wait, the false-condition? And the false-condition is `false` but wait what's the true-condition? A number from strpos() or false?
Or?
😣 
Sigh, I suck at programming."

I'd like using an instance variable to make the return statement easier to read if we were doing more complex true/false work but in the ternary operator, but in this case I think all these issues are solved by simply returning the result of comparing the strpos return value:

    private function containsValue()
    {
        return false !== strpos($this->str, $this->value);
    }

What do you think?